frame/contracts-registry/src/lib.rs

```diff
@@ -0,0 +1,317 @@
+		NotJudge,
+		/// Caller isn't a party
+		NotParties,
+	}
+
+	#[pallet::event]
+	#[pallet::generate_deposit(pub(super) fn deposit_event)]
+	pub enum Event<T: Config<I>, I: 'static = ()> {
+		/// A judge has been added.
+		AddedJudge { judge_id: ContractIndex, deposit: T::AccountId },
+		/// A judge has been removed.
+		RemovedJudge { judge: T::AccountId },
+		/// Judge signed a contract
+		JudgeSigned { contract_id: ContractIndex, signer: T::AccountId },
+		/// Judge signed a contract
+		PartieSigned { contract_id: ContractIndex, signer: T::AccountId },
+		/// Created new contract
+		ContractCreated { contract_id: ContractIndex, creator: T::AccountId },
+	}
+
+	/// Contracts storage containing a content of contracts
+	///
+	/// TWOX-NOTE: Safe, as increasing integer keys are safe.
+	#[pallet::storage]
+	#[pallet::getter(fn contracts)]
+	pub type Contracts<T: Config<I>, I: 'static = ()> = StorageMap<
+		_,
+		Twox64Concat,
+		ContractIndex,
+		ContractDataStorage<T::MaxContractContentLen, T::MaxSignatures, T::AccountId>,
+		OptionQuery,
+	>;
+
+	/// Vec of parties signatures for each contract
+	///
+	/// TWOX-NOTE: Safe, as increasing integer keys are safe.
+	#[pallet::storage]
+	#[pallet::getter(fn parties_signatures)]
+	pub type PartiesSignatures<T: Config<I>, I: 'static = ()> = StorageMap<
+		_,
+		Twox64Concat,
+		ContractIndex,
+		BoundedVec<T::AccountId, T::MaxSignatures>,
+		OptionQuery,
+	>;
+
+	/// Vec of judge signatures for each contract
+	///
+	/// TWOX-NOTE: Safe, as increasing integer keys are safe.
+	#[pallet::storage]
+	#[pallet::getter(fn judge_signatures)]
+	pub(super) type JudgeSignatures<T: Config<I>, I: 'static = ()> = StorageMap<
+		_,
+		Twox64Concat,
+		ContractIndex,
+		BoundedVec<T::AccountId, T::MaxSignatures>,
+		OptionQuery,
+	>;
+
+	/// List of judges
+	#[pallet::storage]
+	#[pallet::getter(fn judges)]
+	pub type Judges<T: Config<I>, I: 'static = ()> =
+		StorageValue<_, BoundedVec<T::AccountId, T::MaxJudges>, ValueQuery>;
+
+	/// The number of contracts that have been made so far.
+	#[pallet::storage]
+	#[pallet::getter(fn contracts_count)]
+	pub type ContractsCount<T: Config<I>, I: 'static = ()> =
+		StorageValue<_, ContractIndex, ValueQuery>;
+
+	#[pallet::call]
+	impl<T: Config<I>, I: 'static> Pallet<T, I> {
+		/// Add judge
+		///
+		/// The dispatch origin of this call must be _Signed_ and the sender must
+		/// have funds to cover the deposit.
+		///
+		/// - `judge`: Account that will be added as one of a judges
+		///
+		/// Emits `AddedJudge`.
+		#[pallet::call_index(0)]
+		#[pallet::weight(T::WeightInfo::add_judge())]
+		pub fn add_judge(origin: OriginFor<T>, judge: T::AccountId) -> DispatchResult {
+			let _ = T::AddJudgeOrigin::ensure_origin(origin)?;
+
+			let real_judges_count = Judges::<T, I>::decode_len().unwrap_or(0) as u32;
+			Judges::<T, I>::try_append(&judge).map_err(|_| Error::<T, I>::TooMany)?;
+
+			Self::deposit_event(Event::AddedJudge {
+				judge_id: real_judges_count + 1,
```

Comment:
```suggestion
				judge,
```
`judge_id` as an integer doesn't seem to have any meaning anywhere in this contract. Do we need it? If not let's just put `judge` with T::AccountId in it.